pkg/store/consul/consulutil/watch_test.go

```diff
@@ -207,6 +208,65 @@ func TestWatchPrefix(t *testing.T) {
 	}
 }
 
+// TestWatchKeys verifies some basic operations of the WatchKeys() function. It
+// should find existing keys, send new updates when the data changes, and
+// ignore changes outside its prefix.
+func TestWatchKeys(t *testing.T) {
+	t.Parallel()
+	f := NewFixture(t)
+	defer f.Stop()
+
+	done := make(chan struct{})
+	defer func() {
+		if done != nil {
+			close(done)
+		}
+	}()
+	kv1a := &api.KVPair{Key: "prefix/hello", Value: []byte("these")}
+	kv1b := &api.KVPair{Key: "prefix/hello", Value: []byte("do")}
+	kv2a := &api.KVPair{Key: "prefix/test", Value: []byte("not")}
+	kv3a := &api.KVPair{Key: "something", Value: []byte("matter")}
+
+	// Process existing data
+	f.Client.KV().Put(kv1a, nil)
+	keysChan := WatchKeys("prefix/", f.Client.KV(), done, 0)
+	keys := <-keysChan
+	if !reflect.DeepEqual(keys.Keys, []string{kv1a.Key}) {
+		t.Error("existing data not recognized")
```

Comment:
optional: perhaps the keys that do exist can be shown, to aid debugging if something does fail in tests.